src/sentry/notifications/utils/participants.py

```diff
@@ -85,29 +85,19 @@ def get_participants_for_group(
     return participants_by_provider
 
 
-def get_reason(
-    user: User, value: NotificationSettingOptionValues, user_ids: set[int]
-) -> int | None:
-    # Members who opt into all deploy emails.
-    if value == NotificationSettingOptionValues.ALWAYS:
-        return GroupSubscriptionReason.deploy_setting
-
-    # Members which have been seen in the commit log.
-    elif value == NotificationSettingOptionValues.COMMITTED_ONLY and user.id in user_ids:
-        return GroupSubscriptionReason.committed
-    return None
-
-
 def get_participants_for_release(
-    projects: Iterable[Project], organization: Organization, user_ids: set[int]
+    projects: Iterable[Project],
+    organization: Organization,
+    user_ids: set[int],
+    notification_setting_type=NotificationSettingTypes.DEPLOY,
```

Comment:
Introduced this optional field to differentiate the `NotificationSettingTypes`. Before, this method was only being called in DeployNotifications and the impl assumes as much. Since we're changing the `ReleaseSummaryActivityNotification.notification_setting_type` to be something besides `NotificationSettingTypes.DEPLOY`, we must parameterize this method to handle the extra case.